### ISSUE TITLE
Button: aria-invalid border only works on outline variant in light mode

### ISSUE THREAD

**Issue by Blackwidow-sudo** [huntabyte/shadcn-svelte#2251]
### Describe the bug

Currently, when you pass the `aria-invalid` prop, it doesn't have any effect on the style of the button even though `aria-invalid:border-destructive` is used in the registry template:
https://github.com/huntabyte/shadcn-svelte/blob/fb24c5d7085efb303094361042fa0049638d0b7f/registry-template/src/lib/registry/ui/button/button.svelte#L4-L16
There are two issues here:
1. Only the outline variant has a `border`, so all other variants dont render any border.
2. If you use the outline variant tho, you still dont get a red-border in dark mode because it gets overwritten by `dark:border-input`

Same issue with the `aria-invalid:ring-destructive`.

<img width="261" height="322" alt="Image" src="https://github.com/user-attachments/assets/3a667450-943a-4489-9d8f-6705738e5d0c" />

I would expect them to look like this (excluding the destructive variant maybe):

<img width="272" height="322" alt="Image" src="https://github.com/user-attachments/assets/0a2c42dc-a177-42a4-b147-9f35ba7576e6" />

### Reproduction

For easy reproduction you can use [my repo](https://github.com/Blackwidow-sudo/shadcn-playground).
I just rendered all button variants in light and dark mode with the `aria-invalid` attribute.

### Logs

```bash
None
```

### System Info

```bash
System:
    OS: Linux 6.6 Ubuntu 24.04.2 LTS 24.04.2 LTS (Noble Numbat)
    CPU: (16) x64 AMD Ryzen 7 7700X 8-Core Processor
    Memory: 8.93 GB / 15.17 GB
    Container: Yes
    Shell: 5.2.21 - /bin/bash
  Binaries:
    Node: 22.17.1 - ~/.local/share/pnpm/node
    npm: 10.9.2 - ~/.local/share/pnpm/npm
    pnpm: 10.13.1 - ~/.local/share/pnpm/pnpm
  npmPackages:
    @lucide/svelte: ^0.515.0 => 0.515.0 
    @sveltejs/kit: ^2.22.0 => 2.26.1 
    bits-ui: ^2.9.0 => 2.9.0 
    svelte: ^5.0.0 => 5.37.1 
    tailwindcss: ^4.0.0 => 4.1.11
```

### Severity

annoyance

**Comment by Blackwidow-sudo** [comment]
Any update on this?
Im using the `DatePicker` component which internally uses the `Button` component. That means that `aria-invalid` has no visual effect on the `DatePicker`.

**Comment by ieedan** [comment]
Hm it's easy enough to fix you just need to apply a ring when the button is marked invalid like:
```
aria-invalid:ring-1
```

But these styles DO match the original and I'm not sure what exactly the intention is from shadcn. Cause the ring would still be applied if you were to focus the button. I guess the best thing to do would be to open up an issue upstream and see if that's the intention or not...

**Comment by Blackwidow-sudo** [comment]
> Hm it's easy enough to fix you just need to apply a ring when the button is marked invalid like:
> 
> ```
> aria-invalid:ring-1
> ```

If you mean like:
```svelte
<Button aria-invalid class="aria-invalid:ring-1">
  2025-03-04
</Button>
```
then no, that doesnt work. You would have to change the variants of the `Button` component itself because it gets overridden by chance (depends on the order of the tailwind output i guess).

<img width="356" height="324" alt="Image" src="https://github.com/user-attachments/assets/2cc2a81a-1f01-4f53-9938-89d63dbc1dc5" />

<img width="352" height="233" alt="Image" src="https://github.com/user-attachments/assets/cab4fe8d-34f3-40f9-b71e-99e47c1c75f4" />

But yeah i will open an issue on upstream shadcn to see what the intention is. Because either show it on all or none. Currently it only works on `variant="outline"` in light mode.

**Comment by ieedan** [comment]
It looks like this will be fixed with #2471

**Comment by Blackwidow-sudo** [comment]
there is also this PR https://github.com/shadcn-ui/ui/pull/8999
